Mage.Sets/src/mage/cards/h/HelgaSkittishSeer.java

```diff
@@ -0,0 +1,165 @@
+import mage.abilities.common.SpellCastControllerTriggeredAbility;
+import mage.abilities.costs.Cost;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.dynamicvalue.DynamicValue;
+import mage.abilities.dynamicvalue.common.SourcePermanentPowerCount;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.abilities.effects.common.GainLifeEffect;
+import mage.abilities.effects.common.counter.AddCountersSourceEffect;
+import mage.abilities.effects.mana.ManaEffect;
+import mage.abilities.mana.SimpleManaAbility;
+import mage.abilities.mana.builder.ConditionalManaBuilder;
+import mage.abilities.mana.conditional.CreatureCastManaCondition;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.choices.ChoiceColor;
+import mage.constants.*;
+import mage.counters.CounterType;
+import mage.filter.FilterSpell;
+import mage.filter.common.FilterCreatureSpell;
+import mage.filter.predicate.mageobject.ManaValuePredicate;
+import mage.game.Game;
+import mage.players.Player;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+public class HelgaSkittishSeer extends CardImpl {
+
+    private static final FilterSpell filter = new FilterCreatureSpell("a creature spell with mana value 4 or greater");
+
+    static {
+        filter.add(new ManaValuePredicate(ComparisonType.OR_GREATER, 4));
+    }
+
+    public HelgaSkittishSeer(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{G}{W}{U}");
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.FROG);
+        this.subtype.add(SubType.DRUID);
+        this.power = new MageInt(1);
+        this.toughness = new MageInt(3);
+
+        // Whenever you cast a creature spell with mana value 4 or greater, you draw a card, gain 1 life and put a +1/+1 counter on Helga, Skittish Seer
+        Ability ability = new SpellCastControllerTriggeredAbility(new DrawCardSourceControllerEffect(1).
+                setText("you draw a card"),
+                filter, false);
+        ability.addEffect(new GainLifeEffect(1).setText("gain 1 life").concatBy(","));
+        ability.addEffect(new AddCountersSourceEffect(CounterType.P1P1.createInstance())
+                .setText("and put a +1/+1 counter on {this}"));
+
+        this.addAbility(ability);
+
+        // {T}: Add X mana of any one color, where X is Helga, Skittish Seer's power. Use this mana only to cast creature spells with mana value 4 or greater or to cas creature spells with {x} in their mana costs
+        this.addAbility(
+                new SimpleManaAbility(
+                        Zone.BATTLEFIELD,
+                        new HelgaSkittishSeerManaEffect(new SourcePermanentPowerCount()),
+                        new TapSourceCost()
+                )
+        );
+    }
+
+    private HelgaSkittishSeer(final HelgaSkittishSeer card) {super(card);}
```

Comment:
style: newline after opening brace and before closing brace, even for boilerplate